src/org/opendatakit/briefcase/ui/ScrollingStatusListDialog.java

```diff
@@ -128,8 +130,12 @@ private ScrollingStatusListDialog(Frame frame, String labelText, String title, I
   public void onEvent(FormStatusEvent event) {
     // Since there can be multiple FormStatusEvent's published concurrently,
     // we have to check if the event is meant for this dialog instance.
-    if (event.getStatus().getFormDefinition().equals(form)) {
-      editorArea.setText(event.getStatus().getStatusHistory());
+    if (isShowing() && event.getStatus().getFormDefinition().equals(form)) {
+      try {
+        editorArea.read(new StringReader(event.getStatus().getStatusHistory()), null);
```

Comment:
That works well but it still seems unfortunate to need to build the model every time! Additionally, it means the scroll position doesn't update as more status messages are added -- I believe some users watch this live and the scrolling is convenient in that context.

What about something like:
```
      try {
        Document doc = editorArea.getDocument();
        String latestStatus = event.getStatus().getStatusString() + "\n";
        doc.insertString(doc.getLength(), latestStatus, null);
      } catch (BadLocationException e) {
        log.warn("Couldn't append status", e);
      }
```

I haven't profiled this so I may be wrong but I _think_ it will also be fast.